src/Microsoft.Azure.SignalR.Common/ClientInvocation/ClientResultsManager.cs

```diff
@@ -0,0 +1,312 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#if NET7_0_OR_GREATER
+using System.Threading;
+using System.Diagnostics;
+using System.Collections.Generic;
+using System.Collections.Concurrent;
+using System.Threading.Tasks;
+using System.Buffers;
+using System;
+using Microsoft.Azure.SignalR.Protocol;
+using Microsoft.AspNetCore.SignalR.Protocol;
+using Microsoft.AspNetCore.SignalR;
+
+namespace Microsoft.Azure.SignalR
+{
+    internal class ClientResultsManager : IClientResultsManager, IInvocationBinder
+    {
+        private readonly ConcurrentDictionary<string, (Type Type, string ConnectionId, object Tcs, Action<object, CompletionMessage> Complete)> _pendingInvocations = new();
+        private ulong _lastInvocationId = new();
+
+        private readonly ConcurrentDictionary<string, (Type Type, string ConnectionId, string CallerServerId, object Tcs, Action<object, CompletionMessage> Complete)> _routedInvocations = new();
+
+        private readonly ConcurrentDictionary<string, ServiceMappingMessage> _serviceMappingMessages = new();
+        private readonly IHubProtocolResolver _hubProtocolResolver;
+
+        public ClientResultsManager(IHubProtocolResolver hubProtocolResolver)
+        {
+            _hubProtocolResolver = hubProtocolResolver;
+        }
+
+        public ulong GetNewInvocation()
```

Comment:
is exposing this needed?

Reply:
used in [src/Microsoft.Azure.SignalR/HubHost/ServiceLifetimeManagerBase.cs](https://github.com/Azure/azure-signalr/pull/1687/files#diff-421d2ddd4830ecaab4e6abbccef7751e56cec4eea3e9c0dec81f2a00cc1a1268R306)